viite-backend/viite-main/src/main/scala/fi/liikennevirasto/viite/process/strategy/TerminatedTwoTrackSectionSynchronizer.scala

```diff
@@ -0,0 +1,473 @@
+    }
+
+    val processedLinks: Seq[ProjectLink] = {
+
+      if (projectLinks.exists(_.status == RoadAddressChangeType.New))
+        throw ViiteException(s"New links are not allowed for the process of adjusting terminated two track links!")
+
+      // Check that there are terminated links on both tracks
+      if (existsTerminationsOnParallelTracks(projectLinks)) {
+        val orderedProjectLinks = projectLinks.sortBy(_.addrMRange.start)
+
+        // List of termination cases
+        val terminationCases: List[Seq[ProjectLink] => Seq[ProjectLink]] = List(
+          roadPartStartTerminated,
+          roadPartEndTerminated,
+          roadPartMiddleTerminated
+        )
+
+        // Sequentially apply each case function
+        terminationCases.foldLeft(orderedProjectLinks)((linksAccumulator, caseFunction) => caseFunction(linksAccumulator))
+
+      } else {
+        // else we return the links unchanged.
+        projectLinks
+      }
+    }
+    processedLinks
+  }
+
+  /**
+   * Takes in project links with same RoadAddressChangeType and divides them in to continuous sections.
+   *
+   * Continuous section has same roadPart, change type / status, track and each project link starts from the
+   * same addrM where the previous project link ended.
+   *
+   * @param projectLinksWithSameStatus Sequence of project links that all share the same status / change type.
+   * @param sectionStatus RoadAddressChangeType i.e. the change type / status of the project links.
+   */
+  def toContinuousSectionsByStatus(projectLinksWithSameStatus: Seq[ProjectLink], sectionStatus: RoadAddressChangeType): Seq[Seq[ProjectLink]] = {
+    var sections = Seq.empty[Seq[ProjectLink]]
+    var currentSection = Seq.empty[ProjectLink]
+    for (link <- projectLinksWithSameStatus) {
+      if (currentSection.isEmpty) {
+        // Start a new section with the given status
+        if (link.status == sectionStatus) {
+          currentSection :+= link
+        }
+      } else {
+        // Check if the current link continues the section
+        val lastLink = currentSection.last
+        if (link.roadPart == lastLink.roadPart &&
+          link.status == sectionStatus &&
+          lastLink.addrMRange.continuesTo(link.addrMRange) &&
+          lastLink.track == link.track) {
+          currentSection :+= link
+        } else {
+          // If it doesn't match, finalize the current section and start a new one
+          sections :+= currentSection
+          currentSection = Seq.empty
+          if (link.status == sectionStatus) {
+            currentSection :+= link
+          }
+        }
+      }
+    }
+    // Add the last section if it exists
+    if (currentSection.nonEmpty)
+      sections :+= currentSection
+    sections
+  }
+
+  /**
+   * Adjusts terminated links to match at both ends by averaging the start- and endAddrMs' of the project links.
```

Comment:
* Adjusts terminated links to match at both ends by averaging the start- and endAddrMs' of the project links.
  ->
   * Adjusts terminated links to have matching averaged start, and end addresses.
  ?